src/components/scenes/newAccount/NewAccountUsernameScene.tsx

```diff
@@ -4,56 +4,116 @@ import { KeyboardAwareScrollView } from 'react-native-keyboard-aware-scroll-view
 import { cacheStyles } from 'react-native-patina'
 import { sprintf } from 'sprintf-js'
 
-import {
-  checkUsernameForAvailabilty,
-  validateUsername
-} from '../../../actions/CreateAccountActions'
+import { fetchIsUsernameAvailable } from '../../../actions/CreateAccountActions'
 import s from '../../../common/locales/strings'
 import { useHandler } from '../../../hooks/useHandler'
 import { Branding } from '../../../types/Branding'
-import { Dispatch, RootState } from '../../../types/ReduxTypes'
-import { connect } from '../../services/ReduxStore'
-import { Theme, ThemeProps, withTheme } from '../../services/ThemeContext'
+import { Dispatch, useDispatch } from '../../../types/ReduxTypes'
+import { logEvent } from '../../../util/analytics'
+import { Theme, useTheme } from '../../services/ThemeContext'
 import { EdgeText } from '../../themed/EdgeText'
 import { MainButton } from '../../themed/MainButton'
 import { OutlinedTextInput } from '../../themed/OutlinedTextInput'
 import { ThemedScene } from '../../themed/ThemedScene'
 
-interface OwnProps {
+interface Props {
   branding: Branding
+  onBack?: (() => void) | undefined
 }
-interface StateProps {
-  username: string
-  usernameErrorMessage: string | null
-}
-interface DispatchProps {
-  checkUsernameForAvailabilty: (username: string) => Promise<void>
-  validateUsername: (username: string) => void
-  onBack: () => void
-}
-type Props = OwnProps & StateProps & DispatchProps & ThemeProps
-
-const NewAccountUsernameSceneComponent = ({
-  theme,
-  onBack,
-  branding,
-  username,
-  usernameErrorMessage,
-  checkUsernameForAvailabilty,
-  validateUsername
-}: Props) => {
+
+const AVAILABILITY_CHECK_DELAY_MS = 1000
+
+type Timeout = ReturnType<typeof setTimeout>
+
+export const NewAccountUsernameScene = ({ branding, onBack }: Props) => {
+  const theme = useTheme()
   const styles = getStyles(theme)
+  const dispatch = useDispatch()
+
+  const mounted = React.useRef<boolean>(true)
+  const [username, setUsername] = React.useState('')
+  const [timerId, setTimerId] = React.useState<Timeout | undefined>(undefined)
+  const fetchCounter = React.useRef<number>(0)
+  const [availableText, setAvailableText] = React.useState<string | undefined>(
+    undefined
+  )
+  const [errorText, setErrorText] = React.useState<string | undefined>(
+    undefined
+  )
+  const [isFetchingAvailability, setIsFetchingAvailability] = React.useState(
+    false
+  )
+
+  React.useEffect(
+    () => () => {
+      mounted.current = false
+    },
+    []
+  )
+
+  const fetchIsAvailable = async (username: string): Promise<boolean> => {
+    return await dispatch(fetchIsUsernameAvailable(username))
+  }
+
+  // Special considerations for the call to action (next) button:
+  // 1. We want to avoid making the user wait for a fetch to check username
+  //    availability if the fetch was already done prior to pressing the button.
```

Comment:
My original implementation supported this - making the next button available
before fetching, saving the user from waiting on the
AVAILABILITY_CHECK_DELAY_MS.